Race condition in SmallRye Config property expansion for @RolesAllowed value?

We sometimes have `LazyAuthRolesAllowedConfigExpTestCase` failing on CI with the error above.
I saw it on Windows which is usually a bit slower than the Linux hosts so it could be a race condition somewhere.

```
2023-02-07T16:47:39.2239103Z [ERROR] io.quarkus.resteasy.reactive.server.test.security.LazyAuthRolesAllowedConfigExpTestCase  Time elapsed: 1.229 s  <<< ERROR!
2023-02-07T16:47:39.2240204Z java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus
2023-02-07T16:47:39.2241332Z 	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:689)
2023-02-07T16:47:39.2242144Z 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$12(ClassBasedTestDescriptor.java:395)
2023-02-07T16:47:39.2243041Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2023-02-07T16:47:39.2243964Z 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:395)
2023-02-07T16:47:39.2244876Z 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:211)
2023-02-07T16:47:39.2245670Z 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:84)
2023-02-07T16:47:39.2246476Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
2023-02-07T16:47:39.2247297Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2023-02-07T16:47:39.2248130Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
2023-02-07T16:47:39.2248832Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2023-02-07T16:47:39.2249559Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
2023-02-07T16:47:39.2250371Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2023-02-07T16:47:39.2251227Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
2023-02-07T16:47:39.2252249Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
2023-02-07T16:47:39.2252872Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
2023-02-07T16:47:39.2253772Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
2023-02-07T16:47:39.2255230Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
2023-02-07T16:47:39.2256141Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2023-02-07T16:47:39.2256993Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
2023-02-07T16:47:39.2257724Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
2023-02-07T16:47:39.2258448Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
2023-02-07T16:47:39.2259250Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2023-02-07T16:47:39.2260108Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
2023-02-07T16:47:39.2260920Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
2023-02-07T16:47:39.2262051Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
2023-02-07T16:47:39.2263172Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
2023-02-07T16:47:39.2264096Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
2023-02-07T16:47:39.2264979Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
2023-02-07T16:47:39.2265807Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
2023-02-07T16:47:39.2266657Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
2023-02-07T16:47:39.2267520Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
2023-02-07T16:47:39.2268531Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
2023-02-07T16:47:39.2269456Z 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
2023-02-07T16:47:39.2270180Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
2023-02-07T16:47:39.2270826Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
2023-02-07T16:47:39.2271554Z 	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
2023-02-07T16:47:39.2272297Z 	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:55)
2023-02-07T16:47:39.2273054Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:223)
2023-02-07T16:47:39.2273930Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:175)
2023-02-07T16:47:39.2274788Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:139)
2023-02-07T16:47:39.2275526Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
2023-02-07T16:47:39.2276189Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
2023-02-07T16:47:39.2276776Z 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
2023-02-07T16:47:39.2277356Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
2023-02-07T16:47:39.2277842Z Caused by: java.lang.RuntimeException: Failed to start quarkus
2023-02-07T16:47:39.2278296Z 	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
2023-02-07T16:47:39.2278774Z 	at io.quarkus.runtime.Application.start(Application.java:101)
2023-02-07T16:47:39.2279307Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-02-07T16:47:39.2280073Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2023-02-07T16:47:39.2280831Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-02-07T16:47:39.2281453Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2023-02-07T16:47:39.2281962Z 	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:253)
2023-02-07T16:47:39.2282529Z 	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:642)
2023-02-07T16:47:39.2282905Z 	... 43 more
2023-02-07T16:47:39.2283938Z Caused by: java.util.NoSuchElementException: SRCFG00014: The config property QuarkusSecurityRolesAllowedConfigSource.property-0 is required but it could not be found in any config source
2023-02-07T16:47:39.2284684Z 	at io.smallrye.config.SmallRyeConfig.convertValue(SmallRyeConfig.java:298)
2023-02-07T16:47:39.2285230Z 	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:242)
2023-02-07T16:47:39.2285849Z 	at io.smallrye.config.SmallRyeConfig.getValue(SmallRyeConfig.java:168)
2023-02-07T16:47:39.2286414Z 	at io.quarkus.security.runtime.SecurityCheckRecorder$1.get(SecurityCheckRecorder.java:75)
2023-02-07T16:47:39.2287047Z 	at io.quarkus.security.runtime.SecurityCheckRecorder$1.get(SecurityCheckRecorder.java:66)
2023-02-07T16:47:39.2287907Z 	at io.quarkus.security.runtime.interceptor.check.SupplierRolesAllowedCheck.resolveAllowedRoles(SupplierRolesAllowedCheck.java:43)
2023-02-07T16:47:39.2288933Z 	at io.quarkus.security.runtime.SecurityCheckRecorder.resolveRolesAllowedConfigExpRoles(SecurityCheckRecorder.java:105)
2023-02-07T16:47:39.2289775Z 	at io.quarkus.deployment.steps.SecurityProcessor$resolveConfigExpressionRoles1389290116.deploy_0(Unknown Source)
2023-02-07T16:47:39.2290511Z 	at io.quarkus.deployment.steps.SecurityProcessor$resolveConfigExpressionRoles1389290116.deploy(Unknown Source)
2023-02-07T16:47:39.2290982Z 	... 51 more
```

--- a/extensions/security/deployment/src/main/java/io/quarkus/security/deployment/SecurityProcessor.java
+++ b/extensions/security/deployment/src/main/java/io/quarkus/security/deployment/SecurityProcessor.java
@@ -46,6 +46,7 @@ import io.quarkus.builder.item.MultiBuildItem;
 import io.quarkus.deployment.Feature;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.Consume;
 import io.quarkus.deployment.annotations.ExecutionTime;
 import io.quarkus.deployment.annotations.Record;
 import io.quarkus.deployment.builditem.ApplicationClassPredicateBuildItem;
@@ -54,6 +55,7 @@ import io.quarkus.deployment.builditem.GeneratedNativeImageClassBuildItem;
 import io.quarkus.deployment.builditem.LaunchModeBuildItem;
 import io.quarkus.deployment.builditem.NativeImageFeatureBuildItem;
 import io.quarkus.deployment.builditem.RunTimeConfigBuilderBuildItem;
+import io.quarkus.deployment.builditem.RuntimeConfigSetupCompleteBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.JPMSExportBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.NativeImageSecurityProviderBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
@@ -536,6 +538,7 @@ public class SecurityProcessor {
                         }).done());
     }
 
+    @Consume(RuntimeConfigSetupCompleteBuildItem.class)
     @BuildStep
     @Record(ExecutionTime.RUNTIME_INIT)
     public void resolveConfigExpressionRoles(Optional<ConfigExpRolesAllowedSecurityCheckBuildItem> configExpRolesChecks,

